test/e2e/specs/editor/plugins/iframed-block.spec.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'changing image size', () => {
+	test.beforeEach( async ( { requestUtils, admin } ) => {
+		await requestUtils.activatePlugin( 'gutenberg-test-iframed-block' );
+		await admin.createNewPost( { postType: 'page' } );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deactivatePlugin( 'gutenberg-test-iframed-block' );
+	} );
+
+	test( 'Should save the changes', async ( { editor, page } ) => {
+		await editor.insertBlock( { name: 'test/iframed-block' } );
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+
+		await expect(
+			page.locator( 'role=document[name="Block: Iframed Block"i]' )
+		).toHaveText( 'Iframed Block (set with jQuery)' );
+
+		// open page from sidebar settings
+		await page.click( '[aria-label="Page"]' );
+
+		// Opens the template editor with a newly created template.
+		await page.click( 'role=button[name="Select template"i]' );
+		await page.click( 'role=button[name="Add template"i]' );
+		await page.fill( 'role=textbox[name="NAME"i]', 'Test template' );
+		await page.click( 'role=button[name="Create"i]' );
+
+		// Expect iframe canvas to be visible
+		await expect(
+			page.locator( 'iframe[name="editor-canvas"]' )
+		).toBeVisible();
+
+		// Expect the script to load in the iframe, which replaces the block text.
+		const iframedText = page.frameLocator( 'iframe' ).locator( 'p' );
```

Comment:
On second thoughts, it seems like the original selector uses the classname selector (`.wp-block-test-iframed-block`) because it's coming from the test plugin which we have total control of. It's better to use `data-testid` though for clearer intention though.

Reply:
@kevin940726 I have used `(locator( 'body' )`  instead of `p`. Also i have tried with `data-testid` but seems it's not working for this case. However the `(locator( 'body' )` approach seems to be working fine. Thanks for your feedbacks!

Reply:
We would need to add the `data-testid="iframed-block-text"` attribute to the plugin code to enable using that approach.